### PR TITLE
fix(core): resurrect OnChange interface

### LINKED ISSUE #2087: Resurrect OnChange
It is required for Dart. It has to compile to:

```dart
abstract class OnChange {
  void onChange(changes);
}
```

### PR DESCRIPTION
**Pull request by yjbanov**
Fixes #2087

### CODE REVIEW

File: modules/angular2/src/core/compiler/interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import {Map} from 'angular2/src/facade/collection';
+
+/**
+ * Defines lifecycle method [onChange] called after all of component's bound
+ * properties are updated.
+ */
+export interface OnChange { onChange(changes: Map<any, any>): void; }
```
Comment (vsavkin) [inline modules/angular2/src/core/compiler/interfaces.ts]: I believe it should be `StringMap<string, any>`

File: modules/angular2/test/core/compiler/integration_dart_spec.dart [hunk 2]
```diff
@@ -104,6 +104,24 @@ main() {
       });
     }));
   });
+
+  describe('OnChange', () {
```
Comment (vsavkin) [inline modules/angular2/test/core/compiler/integration_dart_spec.dart]: Why is this test dart only?

File: modules/angular2/src/core/compiler/interfaces.ts (hunk 1 above)
Comment (yjbanov) [inline modules/angular2/src/core/compiler/interfaces.ts]: No such thing in Dart

File: modules/angular2/test/core/compiler/integration_dart_spec.dart (hunk 2 above)
Comment (yjbanov) [inline modules/angular2/test/core/compiler/integration_dart_spec.dart]: See the TODO below. The test will only pass in Dart after https://github.com/angular/angular/issues/2120 is fixed.

File: modules/angular2/src/core/compiler/interfaces.ts (hunk 1 above)
Comment (yjbanov) [inline modules/angular2/src/core/compiler/interfaces.ts]: ts2dart figured it out. Done.

**Comment by yjbanov** [comment]
Got offline LGTM from @vsavkin

**Pull request closed without merging**